tests/unit/test_query.py

```diff
@@ -85,3 +87,29 @@ def test_get_manual_pagination(self):
             headers={"accept": "application/json;"},
             json=None,
         )
+
+
+class AllocationErrorTestCase(unittest.TestCase):
+    @patch(
+        "requests.sessions.Session.get",
+        return_value=Response(fixture="ipam/prefix.json"),
+    )
+    @patch(
+        "requests.sessions.Session.post", return_value=Response(status_code=204),
+    )
+    def test_204_case(self, _, __):
+        nb = pynetbox.api("http://localhost:8000/")
+        with self.assertRaises(AllocationError):
+            _ = nb.ipam.prefixes.get(1).available_ips.create()
+
+    @patch(
+        "requests.sessions.Session.get",
+        return_value=Response(fixture="ipam/prefix.json"),
+    )
+    @patch(
+        "requests.sessions.Session.post", return_value=Response(status_code=409),
+    )
+    def test_409_case(self, _, __):
+        nb = pynetbox.api("http://localhost:8000/")
+        with self.assertRaises(AllocationError):
+            _ = nb.ipam.prefixes.get(1).available_ips.create()
```

Comment:
Trying to get away from the tests that rely on fixtures. Given I want to refactor `_make_call`, I can probably manage to test this in the new setup.